### PR TITLE
635 add auto config transform

### LINKED ISSUE #635: Add auto config transformation to be back-compatible
## Subject of the issue

Recent config changes make existing users confused.
It is therefore important to support previous configs while doing sanity checks.

We can parse, transform and output a new config and suggest to users.

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

Add parser to support config at v0.1.1

Fixes #635

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mathpluscode** [comment]
This PR is ready now @YipengHu @s-sd @NMontanaBrown 

Now we support the v0.1.1 configs.

**Review by YipengHu — APPROVED** [review]
works.

**Comment by NMontanaBrown** [comment]
Will review this this morning @YipengHu @mathpluscode

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,14 +6,15 @@ to make them as straightforward as possible.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project
 adheres to [Semantic Versioning](http://semver.org/).
 
-## [1.0.0] - In Progress
+## [1.0.0rc1] - In Progress
```
Comment (NMontanaBrown) [inline CHANGELOG.md:9]: rc1?

Comment (mathpluscode) [reply CHANGELOG.md:9]: Release candidate 1 :)

Comment (NMontanaBrown) [reply CHANGELOG.md:9]: Strictly speaking according to Semver it sould be 1.0.0-rc1. Why not call it 1.0.0-alpha, see Semver  spec 9: https://semver.org/#spec-item-9

Comment (mathpluscode) [reply CHANGELOG.md:9]: https://www.python.org/dev/peps/pep-0440/#pre-releases

I followed pypip guide PEP 440 :)

Comment (mathpluscode) [reply CHANGELOG.md:9]: For me it's just a format, it can be any one.

Comment (mathpluscode) [reply CHANGELOG.md:9]: the reason of not being alpha because i do not know what will be the name after alpha, beta

Comment (NMontanaBrown) [reply CHANGELOG.md:9]: But if we are following Semver according to the docs as it says in the changelog, it should be 1.0.0-rc.1? So either we change semver to pep440 or we follow Semver.

Comment (mathpluscode) [reply CHANGELOG.md:9]: checking https://github.com/tensorflow/tensorflow/tags so let's go for your suggestion

**Review by NMontanaBrown — APPROVED** [review]
Hi @mathpluscode this is a great piece of work, thanks so much. I would recommend a release with this MR on 0.1.1 or 1.0.0-rc1 (whatever numbr we are on) that pre-empts users to this new config change. What do you think?

**Comment by mathpluscode** [comment]
> Hi @mathpluscode this is a great piece of work, thanks so much. I would recommend a release with this MR on 0.1.1 or 1.0.0-rc1 (whatever numbr we are on) that pre-empts users to this new config change. What do you think?

Yup, I plan to add more docs on registry this week and then let's do a release within one week.

**Pull request merged**